src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -444,6 +445,15 @@ void insert_reorders_in_dir(program& p, const std::map<program_node*, format::ty
             in_layout.format == format::custom || out_layout.format == format::custom)
             continue;
 
+        if (((reinterpret_cast<program_node*>(get_node(next))->is_type<loop>() && dir == direction_e::forwards)
+                || (node->is_type<loop>() && dir == direction_e::backwards))
+                && in_layout.get_rank() != out_layout.get_rank()) {
+            // Don't add reorder if the ranks of prev and node are different.
+            // issue case 1)  scatter_nd_update(bfyx:2x100x4) -> reorder(bfzyx:2x100x4) -> loop(bfzyx:2x100x1x1x360)
+            //                                                   >> added reorder has abnormal format.
+            continue;
```

Comment:
can't we resolve this by adding get_preferred_format in loop?

Reply:
I think we have fundamental issue in layout propagation. When tensor ranks are different, layout propagation just don't make sense.